# HG changeset patch
# Parent ce85866d33cf7807a1b4ce9b2069a14a2c58f084
# User sbr
# Date 1349835632 -7200

Fix (r18325): Make the space between 'current order'-arrows and 'order index' depends of the total number of orders (avoiding overlap if order count is greater than 99).

diff -r ce85866d33cf src/order_gui.cpp
--- a/src/order_gui.cpp	Wed Oct 10 04:01:18 2012 +0200
+++ b/src/order_gui.cpp	Wed Oct 10 05:08:52 2012 +0200
@@ -1097,7 +1097,7 @@
 		if (widget != WID_O_ORDER_LIST) return;
 
 		bool rtl = _current_text_dir == TD_RTL;
-		SetDParam(0, 99);
+		SetDParam(0, this->vehicle->GetNumOrders() < 10 ? 9 : this->vehicle->GetNumOrders() < 100 ? 99 : 999);
 		int index_column_width = GetStringBoundingBox(STR_ORDER_INDEX).width + 2 * GetSpriteSize(rtl ? SPR_ARROW_RIGHT : SPR_ARROW_LEFT).width + 3;
 		int middle = rtl ? r.right - WD_FRAMETEXT_RIGHT - index_column_width : r.left + WD_FRAMETEXT_LEFT + index_column_width;
 
diff -r ce85866d33cf src/timetable_gui.cpp
--- a/src/timetable_gui.cpp	Wed Oct 10 04:01:18 2012 +0200
+++ b/src/timetable_gui.cpp	Wed Oct 10 05:08:52 2012 +0200
@@ -368,7 +368,7 @@
 				bool final_order = false;
 
 				bool rtl = _current_text_dir == TD_RTL;
-				SetDParam(0, 99);
+				SetDParam(0, v->GetNumOrders() < 10 ? 9 : v->GetNumOrders() < 100 ? 99 : 999);
 				int index_column_width = GetStringBoundingBox(STR_ORDER_INDEX).width + 2 * GetSpriteSize(rtl ? SPR_ARROW_RIGHT : SPR_ARROW_LEFT).width + 3;
 				int middle = rtl ? r.right - WD_FRAMERECT_RIGHT - index_column_width : r.left + WD_FRAMERECT_LEFT + index_column_width;
 
